src/HuffmanCode.java

```diff
@@ -48,17 +48,15 @@ public HuffmanCode(Scanner input) {
 		
 	}
 	// Helper Method
+	// pre: code is in a good format
 	// post: creates a new HuffmanCode object by reading a huffman code from a .code file
 	private HuffmanNode helperHuffmanCode(int asciiValue, String code, HuffmanNode root) {	
 		if(code.length() == 0) {
 			return new HuffmanNode((char) asciiValue, 0);
-		} else if (root == null) {
-			root = new HuffmanNode((char) 0, 0);
 		}
 		if(code.substring(0,1).equals("0")) {
 			root.left = helperHuffmanCode(asciiValue, code.substring(1), root.left);
-		}
-		if(code.substring(0,1).equals("1")) {
```

Comment:
As long as you assume that pre: code is in good format, you can use else statement